File: ALifeCommon/src/main/java/com/cristiano/alife/instructions/stack/PushInstruction.java
package com.cristiano.alife.instructions.stack;

import com.cristiano.alife.consts.ALifeConsts;
import com.cristiano.alife.world.BaseInstruction;
import com.cristiano.alife.world.IOrganismo;
import com.cristiano.alife.world.IWorld;

public class PushInstruction extends BaseInstruction  {
	public PushInstruction(IWorld mundo) {
		super(mundo);
	}
	@Override
	public void executa(IOrganismo o) {
		o.push(o.getReg(getByteOrganismo(o,1)));
	}
	
	@Override
	public String getDescription(IOrganismo o,int ip){
		return comment(name+" "+ALifeConsts.getLetter(o.getMemory(ip+1)),"push "+ALifeConsts.getLetter(o.getMemory(ip+1))+"");
	}
	
	@Override
	public int getStep() {
		return 2;
	}
}

File: netAvida/netAvida.Backend/instructions/impl/stack/PushInstruction.cs
using netAvida.backend;
using netAvida.backend.interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace netAvida.Backend.instructions.impl
{
    public class PushInstruction : BaseInstruction
    {
        public PushInstruction(IWorld mundo): base(mundo)
        {
        }

    public override void executa(IOrganismo o)
        {
            o.push(o.getReg(getByteOrganismo(o, 1)));
        }

    public override int getStep()
        {
            return 2;
        }

    public override String getDescription(IOrganismo o, int ip)
        {
            return comment(name + " " + ALifeConsts.getLetter(o.getMemory(ip + 1)), "push " + ALifeConsts.getLetter(o.getMemory(ip + 1)) + "");
        }
    }
}
